pkg/plugins/plugins.go

```diff
@@ -0,0 +1,63 @@
+package plugins
+
+// OpenShiftAzureConfig is a new, Red Hat owned type defined in an external project.  It
+// represents a superset of the configuration required for all steps in the plugin process *after*
+// the manifest is validated.
+//
+// OpenShiftAzureConfig is a versioned type.  It is expected that the configuration will change
+// between releases and may require migration steps in the future.  The storage of this type
+// is opaque to the plugins, they only require that it be passed in.  However, it must be retained
+// to allow it to be lifecycled from version to version.
+type OpenShiftAzureConfig struct{}
+
+// ManagedOpenShiftCluster is the external schema that is passed to the RP.  It is defined in
+// acs-engine.  ManagedOpenShiftCluster will reuse many of the lower level objects shared by
+// ManagedCluster but is separated in order to keep the relationship with AKS clear.
+type ManagedOpenShiftCluster struct{}
```

Comment:
Our current working view is that we believe we need at least:

```
type ManagedOpenShiftCluster struct {
	TenantID               string
	SubscriptionID         string
	ClientID               string
	ClientSecret           string
	Location               string
	ResourceGroup          string
	VMSize                 string
	ComputeCount           int
	InfraCount             int
	RoutingConfigSubdomain string
	PublicHostname         string
}
```

Some of that stuff won't come in explicitly from the external API but will need to be enriched.

Having access to the name of the HCP namespace in which our pods run in would be really useful as it allows us to make our x.509 cert CNs/SANs more specific.